runtime/compiler/runtime/codertinit.cpp

```diff
@@ -222,7 +222,7 @@ J9JITConfig * codert_onload(J9JavaVM * javaVM)
 #elif defined(TR_HOST_64BIT) || defined(TR_HOST_S390)
    // 390 31-bit may generate 64-bit instruction (i.e. CGRL) which requires
    // doubleword alignment for its operands
-   jitConfig->codeCacheAlignment = 8;
+   jitConfig->codeCacheAlignment = 16;
```

Comment:
I think we should limit this change to Z only. @dsouzai @mpirvu FYI regarding this change in AOT alignment.

There is a more general problem in that AOT methods are not being aligned to the same boundary that JIT methods are. I think this is due to not using the same APIs when allocating memory for an AOT load. In normal JIT compilations we will have used `allocateCodeMemory` [1] and a few lines below a call to `alignBinaryBufferCursor` [2] which aligns the entry point based off of `getJitMethodEntryAlignmentBoundary` which codegens override. Every codegen does this in the same way and this was all consolidated in a cross platform way in eclipse/omr#4314.

I don't see AOT code allocations doing the same thing so we can run into various problems as seen in this PR. I think this needs to be addressed and the code being reviewed here needs to be deprecated in favor of the codegen APIs. Thoughts?

[1] https://github.com/eclipse/omr/blob/99d7e521e1f8d083ab6bf558d3b8b9264579e46d/compiler/z/codegen/OMRCodeGenerator.cpp#L2431-L2435
[2] https://github.com/eclipse/omr/blob/99d7e521e1f8d083ab6bf558d3b8b9264579e46d/compiler/codegen/OMRCodeGenerator.cpp#L2196-L2230

Reply:
@fjeremic  In chat I had with @dsouzai , a solution for this issue is to have information about initial alignment in the AOT Method Header so when we start relocation we can use that to get the memory from codecache.

Reply:
Limited alignment changes to Z only in https://github.com/eclipse/openj9/compare/3a4703eaf139c681caf30931dfc1308949d43e15..836d4b19681641d3c250d756ce09c06719a3f5f7